src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2395,24 +2426,68 @@ private static void AddNestedTypesToDictionary(Dictionary<string, ImmutableArray
             }
         }
 
-        private sealed class MembersAndInitializersBuilder
+        protected sealed class MembersAndInitializersBuilder
         {
-            public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; } = ArrayBuilder<Symbol>.GetInstance();
-            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> StaticInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
-            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> InstanceInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
-            public readonly ArrayBuilder<SyntaxReference> IndexerDeclarations = ArrayBuilder<SyntaxReference>.GetInstance();
+            public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; }
+            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> StaticInitializers;
+            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> InstanceInitializers;
+            public readonly ArrayBuilder<SyntaxReference> IndexerDeclarations;
             public RecordDeclarationSyntax? RecordDeclarationWithParameters;
-            public ArrayBuilder<FieldOrPropertyInitializer.Builder>? InstanceInitializersForRecordDeclarationWithParameters;
+            public int InstanceInitializersIndexForRecordDeclarationWithParameters;
```

Comment:
>InstanceInitializersIndexForRecordDeclarationWithParameters [](start = 23, length = 59)

Since it is an index now, it would be good to document where.